### PR TITLE
large test helper refactor

### PR DESCRIPTION
**Pull request by ryanfkeepers**
drive collection tests needed a facelift for readability and ease of production/consumption.  This change is primarily housecleaning with those priorities in mind.  It also contains the support for multi-delta enumerator mocks so that the next PR can focus entirely on testing the multi-delta enum changes.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :robot: Supportability/Tests

#### Issue(s)

* #4689

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/drive/helper_test.go [hunk 1]
```diff
@@ -59,3 +77,768 @@ func loadTestService(t *testing.T) *oneDriveService {
...
+	return NewCollections(
+		mbh,
+		tenant,
+		idname.NewProvider(user, user),
+		func(*support.ControllerOperationStatus) {},
+		opts,
+		count.New())
+}
+
+// func fullOrPrevPath(
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/helper_test.go:567]: Just checking, but what's the plan for this and other commented out code in the file?

File: src/internal/m365/collection/drive/collections_test.go [hunk 2]
```diff
@@ -1719,19 +1412,10 @@ func (suite *CollectionsUnitSuite) TestGet_treeCannotBeUsedWhileIncomplete() {
...
-					},
-				}},
-				DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
-			},
-		},
-	}
+	mbh.DriveItemEnumeration = mock.DriveEnumerator(
+		mock.Drive(id(drive)).With(
+			mock.Delta(id(delta), nil).With(
+				aPage())))
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/collections_test.go:1418]: is this missing the `delItem` that's in the old version of the code?

File: src/internal/m365/collection/drive/collections_tree_test.go [hunk 3]
```diff
@@ -588,11 +338,14 @@ func (suite *CollectionsTreeUnitSuite) TestCollections_PopulateTree_singleDelta(
...
-				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+			enumerator: mock.EnumerateDriveItemsDelta{
+				DrivePagers: map[string]*mock.DriveDeltaEnumerator{
 					id(drive): {
-						Pages:       nil,
-						DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
+						DriveID: id(drive),
+						DeltaQueries: []*mock.DeltaQuery{{
+							Pages:       nil,
+							DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/collections_tree_test.go:347]: nit: can this be swapped with `mock.Delta(id(delta), nil)`?

**Review by ashmrtn — APPROVED** [review]

File: src/internal/m365/collection/drive/helper_test.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/helper_test.go:567]: they get updated here: https://github.com/alcionai/corso/pull/4795/files#diff-48bc25f85f8c885368934862a2813c6fbd7dfb71f309f91169f605230a85895eR656

File: src/internal/m365/collection/drive/collections_test.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/collections_test.go:1418]: good catch!

File: src/internal/m365/collection/drive/collections_tree_test.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/collections_tree_test.go:347]: No can do.  It's a special case that needs an explicit struct creation for the test.  I'll make a note of it.  As for the mock.Delta, it's not the same item type.

**Pull request merged**